public/assets/webworker.js

```diff
@@ -46,14 +46,22 @@ self.addEventListener(
 );
 
 function sendCpuState() {
-  // console.log('webworker responding fetchCpu request');
+  //console.log('webworker responding fetchCpu request');
+  // TODO: add a sendState method for each UI component?
+  //       only worried if the same state will be expressed in each method, due to threading issues
   const payload = {
     type: 'returnCpu',
     d: JSON.stringify({
-        registers: RISCV.gen_reg
+        registers: RISCV.gen_reg,
+        instruction_amounts: RISCV.instruction_amounts,
     }),
   };
   this.postMessage(payload);
+  RISCV.instruction_amounts['arithmetic'] = 0;
```

Comment:
why do we reset every time we send the cpu state?

Reply:
I want to display the number of instructions per every clock cycle, so when we send the cpu state, im resetting it for the next clock cycle

Reply:
what do you mean by clock cycle? do you mean how often a heartbeat happens?

Reply:
yeah, every time it posts a message. So every 1000ms